server-spi-private/src/main/java/org/keycloak/models/UserLoginFailureProviderFactory.java

```diff
@@ -16,11 +16,12 @@
  */
 package org.keycloak.models;
 
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
 import org.keycloak.provider.ProviderFactory;
 
 /**
  * @author <a href="mailto:mkanis@redhat.com">Martin Kanis</a>
  */
-public interface UserLoginFailureProviderFactory<T extends UserLoginFailureProvider> extends ProviderFactory<T>  {
+public interface UserLoginFailureProviderFactory<T extends UserLoginFailureProvider> extends ProviderFactory<T>, EnvironmentDependentProviderFactory {
```

Comment:
I didn't check the previous commits, but now I notice: While this technically works, the interface `EnvironmentDependentProviderFactory` should go on each implementation, as it is not a feature of the `UserLoginFailureProviderFactory`, but of each implementation.